src/Npgsql/PostgresDatabaseInfo.cs

```diff
@@ -197,16 +207,19 @@ internal async Task<List<PostgresType>> LoadBackendTypes(NpgsqlConnector conn, N
         if (timeout.IsSet)
             commandTimeout = (int)timeout.CheckAndGetTimeLeft().TotalSeconds;
 
-        var batchQuery = new StringBuilder()
-            .AppendLine("SELECT version();")
-            .AppendLine(GenerateLoadTypesQuery(SupportsRangeTypes, SupportsMultirangeTypes, conn.Settings.LoadTableComposites))
-            .AppendLine(GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites));
-
-        if (SupportsEnumTypes)
-            batchQuery.AppendLine(GenerateLoadEnumFieldsQuery(HasEnumSortOrder));
+        var versionQuery = "SELECT version();";
+        var loadTypesQuery = GenerateLoadTypesQuery(SupportsRangeTypes, SupportsMultirangeTypes, conn.Settings.LoadTableComposites);
+        var loadCompositeTypesQuery = GenerateLoadCompositeTypesQuery(conn.Settings.LoadTableComposites);
+        var loadEnumFieldsQuery = SupportsEnumTypes
+            ? GenerateLoadEnumFieldsQuery(HasEnumSortOrder)
+            : string.Empty;
 
         timeout.CheckAndApply(conn);
-        await conn.WriteQuery(batchQuery.ToString(), async);
+        await conn.WriteQuery(versionQuery, async);
```

Comment:
Just to note that it should be possible to concatenate in a single Query with semicolons (supported in the simple protocol). Could slightly simplify things, but not important.

Reply:
I'd also add a comment explaining that we're not doing the natural thing (extended protocol + batching) because of logical replication.

Reply:
> Just to note that it should be possible to concatenate in a single Query with semicolons (supported in the simple protocol).

Yeah true. But as you can infer from the comment and the newline problem, logical replication connections don't support the simple protocol but only a subset (it really uses a completely separate lexer and parser).
Sadly this subset also does not support batches but only single commands.

What I'm doing here is actually even a bit dangerous because the database might change between the individual commands leading to errors in the subsequent processing in the results.
But that case seems rare and even a bit constructed and there simply isn't a way to avoid it (at least for replication connections).

> I'd also add a comment explaining that we're not doing the natural thing (extended protocol + batching) because of logical replication.

No Problem. I'll do that.

Reply:
Whatever works is totally fine :) Re possible changes, let's see how this develops in PG, we can always react and fix in a patch if needed.

Reply:
I ended up branching the code for replication and regular connections in a minimal way and implementing `SanitizeForReplicationConnection()` as a simple parsing method that removes comments and carriage returns in one pass.
Since the queries are pretty much constant and hand-crafted at development time and that area is now covered by tests, this feels pretty safe.